x-pack/plugins/alerting/server/application/rule/methods/bulk_delete/bulk_delete.ts

```diff
@@ -8,27 +8,35 @@ import pMap from 'p-map';
 import { KueryNode, nodeBuilder } from '@kbn/es-query';
 import { SavedObjectsBulkUpdateObject } from '@kbn/core/server';
 import { withSpan } from '@kbn/apm-utils';
-import { RawRule } from '../../types';
-import { convertRuleIdsToKueryNode } from '../../lib';
-import { bulkMarkApiKeysForInvalidation } from '../../invalidate_pending_api_keys/bulk_mark_api_keys_for_invalidation';
-import { ruleAuditEvent, RuleAuditAction } from '../common/audit_events';
-import { tryToRemoveTasks } from '../common';
-import { API_KEY_GENERATE_CONCURRENCY } from '../common/constants';
+import { RawRule } from '../../../../types';
+import { convertRuleIdsToKueryNode } from '../../../../lib';
+import { bulkMarkApiKeysForInvalidation } from '../../../../invalidate_pending_api_keys/bulk_mark_api_keys_for_invalidation';
+import { ruleAuditEvent, RuleAuditAction } from '../../../../rules_client/common/audit_events';
+import { tryToRemoveTasks } from '../../../../rules_client/common';
+import { API_KEY_GENERATE_CONCURRENCY } from '../../../../rules_client/common/constants';
 import {
   getAuthorizationFilter,
   checkAuthorizationAndGetTotal,
   getAlertFromRaw,
   migrateLegacyActions,
-} from '../lib';
+} from '../../../../rules_client/lib';
 import {
   retryIfBulkOperationConflicts,
   buildKueryNodeFilter,
-  getAndValidateCommonBulkOptions,
-} from '../common';
-import { BulkOptions, BulkOperationError, RulesClientContext } from '../types';
-
-export const bulkDeleteRules = async (context: RulesClientContext, options: BulkOptions) => {
-  const { ids, filter } = getAndValidateCommonBulkOptions(options);
+} from '../../../../rules_client/common';
+import type { RulesClientContext } from '../../../../rules_client/types';
+import type {
+  BulkOperationErrorV1,
+  BulkDeleteRulesResponseV1,
+  BulkDeleteRulesRequestParamsV1,
+} from '../../../../../common/routes/rule/apis/bulk_delete';
+import { getAndValidateCommonBulkOptionsV1 } from '../../../../../common/routes/rule/apis/bulk_delete';
+
+export const bulkDeleteRules = async (
+  context: RulesClientContext,
+  options: BulkDeleteRulesRequestParamsV1
+): Promise<BulkDeleteRulesResponseV1> => {
+  const { ids, filter } = getAndValidateCommonBulkOptionsV1(options);
```

Comment:
I think a better approach would be to use the schema (`BulkDeleteRulesRequestParams`) to do the validation, similar to how it's done in `create_rule -> line 62` instead of doing custom validation. It helps to keep the sources of truth closer together.

Reply:
2410993ae1d6047cbb272d1af947aa0e8b1acb85

Reply:
I read your docs. And it seems that we need to have 2 files for schema and files, because it's different for application and http layers. 
Why it's different? In my case it's the same. It feels wrong to double the same schema and types.